src/v0/destinations/klaviyo/util.js

```diff
@@ -1,116 +1,75 @@
- * for that user is getting returned else false is returned.
- * Docs: https://developers.klaviyo.com/en/reference/get-profile-id
- * @param {*} message
- * @param {*} Config
- * @returns
+ * This function is used for creating response for subscribing users to a particular list.
+ * DOCS: https://www.klaviyo.com/docs/api/v2/lists
  */
-const isProfileExist = async (message, { Config }) => {
-  const { privateApiKey } = Config;
-  const userIdentifiers = {
+const subscribeUserToList = (message, traitsInfo, destination) => {
+  // listId from message properties are preferred over Config listId
+  const { privateApiKey, consent } = destination.Config;
+  let { listId } = destination.Config;
+  const targetUrl = `${BASE_ENDPOINT}/api/profile-subscription-bulk-create-jobs`;
+  const subscriptionObj = {
     email: getFieldValueFromMessage(message, 'email'),
-    external_id: getFieldValueFromMessage(message, 'userId'),
     phone_number: getFieldValueFromMessage(message, 'phone'),
   };
 
-  // eslint-disable-next-line no-restricted-syntax
-  for (const id in userIdentifiers) {
-    if (isDefinedAndNotNull(userIdentifiers[id])) {
-      // eslint-disable-next-line no-await-in-loop
-      const profileResponse = await httpGET(`${BASE_ENDPOINT}/api/v2/people/search`, {
-        header: {
-          Accept: JSON_MIME_TYPE,
-        },
-        params: {
-          api_key: privateApiKey,
-          [id]: userIdentifiers[id],
-        },
-      });
-      const processedProfileResponse = processAxiosResponse(profileResponse);
-      if (processedProfileResponse.status === 200 && processedProfileResponse.response?.id) {
-        return processedProfileResponse.response.id;
+  if (traitsInfo?.properties?.consent || consent) {
+    const subscribeConsent = traitsInfo?.properties?.consent || consent;
+    const channels = {};
+    if (Array.isArray(subscribeConsent)) {
+      if (subscribeConsent.includes('email')) {
+        channels.email = [...(channels.email || []), 'MARKETING'];
+      }
+      if (subscribeConsent.includes('sms')) {
+        channels.sms = [...(channels.sms || []), 'MARKETING'];
       }
-      if (
-        !(
-          processedProfileResponse.status === 404 &&
-          processedProfileResponse?.response.detail ===
-            'There is no profile matching the given parameters.'
-        )
-      ) {
-        throw new NetworkError(
-          `The lookup call could not be completed with the error:
-          ${JSON.stringify(processedProfileResponse.response)}`,
-          processedProfileResponse.status,
-          {
-            [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(processedProfileResponse.status),
-          },
-          processedProfileResponse,
-        );
+    } else if (subscribeConsent) {
+      if (subscribeConsent === 'sms') {
+        channels.sms = [...(channels.sms || []), 'MARKETING'];
+      } else {
+        channels.email = [...(channels.email || []), 'MARKETING'];
       }
     }
```

Comment:
[Nitpick]
```javascript
// can be an array or string
const subscibeContentArr = subscribeConsent;
if (!Array.isArray(subscribeConsent)) {
    subscribeConsent = [subscribeContent]
}

if (subscribeConsent.includes('email')) {
    channels.email = [...(channels.email || []), 'MARKETING'];
}
if (subscribeConsent.includes('sms')) {
    channels.sms = [...(channels.sms || []), 'MARKETING'];
}

```

I think this looks a lot cleaner. Can we try something like this ?

Reply:
Updated